cpp/models/glct_secir/model.h

```diff
@@ -0,0 +1,364 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef MIO_GLCT_SECIR_MODEL_H
+#define MIO_GLCT_SECIR_MODEL_H
+
+#include "glct_secir/parameters.h"
+#include "glct_secir/infection_state.h"
+#include "memilio/epidemiology/lct_infection_state.h"
+#include "memilio/compartments/compartmentalmodel.h"
+#include "memilio/epidemiology/populations.h"
+#include "memilio/config.h"
+#include "memilio/utils/logging.h"
+#include "memilio/utils/time_series.h"
+#include "memilio/math/eigen.h"
+
+namespace mio
+{
+namespace glsecir
+{
+
+/**
+ * @brief Class that defines an GLCT-SECIR model.
+ *
+ * @tparam NumExposed The number of subcompartments used for the Exposed compartment.
+ * @tparam NumInfectedNoSymptoms The number of subcompartments used for the InfectedNoSymptoms compartment. 
+ * @tparam NumInfectedSymptoms The number of subcompartments used for the InfectedSymptoms compartment.
+ * @tparam NumInfectedSevere The number of subcompartments used for the InfectedSevere compartment.
+ * @tparam NumInfectedCritical The number of subcompartments used for the InfectedCritical compartment.
+ */
+template <size_t NumExposed, size_t NumInfectedNoSymptoms, size_t NumInfectedSymptoms, size_t NumInfectedSevere,
+          size_t NumInfectedCritical>
+class Model
```

Comment:
Can you use LctState as a template parameter analogous to the LCT model?

Reply:
Actually, i had this structure here also for the LCT model before i added age groups. For the case without age groups, i still like this option better as it is less prone to errors (eg. it is not possible to set the numbers of compartments of Susceptible to something different to one). It is easier to just directly set these numbers of subcompartments. This way is not possible for age groups (as we would need parameter packs for all numbers). Therefore, this will be adapted in #1134.